autopatches/patches/new-permission-system/patch.sh

```diff
@@ -0,0 +1,66 @@
+
+cd scripts/
+
+if grep -q 'ynh_legacy_permissions' upgrade || grep -q 'ynh_permission_' install
+then
+    # App already using the new permission system - not patching anything
+    exit 0
+fi
+
+if ! grep -q "protected_\|skipped_" install
+then
+    # App doesn't has any (un)protected / skipped setting ?
+    # Probably not a webapp or permission ain't relevant for it ?
+    exit 0
+fi
+
+CONFIGURE_PERMISSION_DURING_INSTALL='
+# Make app public if necessary
+if [ \"\$is_public\" -eq 1 ]
+then
+	ynh_permission_update --permission=\"main\" --add=\"visitors\"
+fi
+'
+
+MIGRATE_LEGACY_PERMISSIONS='
+#=================================================
+# Migrate legacy permissions to new system
+#=================================================
+if ynh_legacy_permissions_exists
+then
+	ynh_legacy_permissions_delete_all
+
+	ynh_app_setting_delete --app=\$app --key=is_public
+fi'
+
+for SCRIPT in "remove upgrade backup restore change_url"
+do
+    [[ -e $SCRIPT ]] || continue
+
+    perl -p0e 's@.*ynh_app_setting_.*protected_.*@@g' -i $SCRIPT
```

Comment:
just out of curiosity, why `per`l and not `sed`?

Reply:
Ah yes good question haha ... Because the first thing I was working on was finding a way to do a multiline replace, which you can with perl, but not with sed (c.f. the other regex patching "if ...(.|\n) fi") ... but in fact most of these could be done with sed ...